src/autoscaler/metricscollector/collector/app_streamer_test.go

```diff
@@ -102,10 +103,115 @@ var _ = Describe("AppStreamer", func() {
+					msgChan <- noaa.NewHttpStartStopEnvelope(555555, 300000000, 600000000, 1)
+					msgChan <- noaa.NewHttpStartStopEnvelope(666666, 300000000, 700000000, 1)
+				}()
+
+				By("collecting and computing througput and responsetime for second interval")
+				Consistently(database.SaveMetricCallCount).Should(Equal(2))
+
+				By("save throughput and responsetime metric after the second collect interval")
+				fclock.Increment(TestCollectInterval)
+				Eventually(database.SaveMetricCallCount).Should(Equal(4))
+
+				Expect(database.SaveMetricArgsForCall(2)).To(Equal(&models.AppInstanceMetric{
+					AppId:         "an-app-id",
+					InstanceIndex: 1,
+					CollectedAt:   fclock.Now().UnixNano(),
+					Name:          models.MetricNameThroughput,
+					Unit:          models.UnitRPS,
+					Value:         "3.0",
+					Timestamp:     fclock.Now().UnixNano(),
+				}))
+				Expect(database.SaveMetricArgsForCall(3)).To(Equal(&models.AppInstanceMetric{
+					AppId:         "an-app-id",
+					InstanceIndex: 1,
+					CollectedAt:   fclock.Now().UnixNano(),
+					Name:          models.MetricNameResponseTime,
+					Unit:          models.UnitMilliseconds,
+					Value:         "300",
+					Timestamp:     fclock.Now().UnixNano(),
+				}))
+			})
+
+			Context("when the app has multiple instances", func() {
+				BeforeEach(func() {
+					go func() {
+						msgChan <- noaa.NewHttpStartStopEnvelope(111111, 100000000, 200000000, 0)
+						msgChan <- noaa.NewHttpStartStopEnvelope(222222, 300000000, 500000000, 1)
+						msgChan <- noaa.NewHttpStartStopEnvelope(333333, 200000000, 600000000, 2)
+						msgChan <- noaa.NewHttpStartStopEnvelope(555555, 300000000, 500000000, 2)
+					}()
+
+				})
+				It("saves throughput and responsetime metrics of multiple instances", func() {
+					Consistently(database.SaveMetricCallCount).Should(Equal(0))
+					fclock.WaitForWatcherAndIncrement(TestCollectInterval)
+					Eventually(database.SaveMetricCallCount).Should(Equal(6))
+				})
+			})
+
+			Context("when database fails", func() {
+				BeforeEach(func() {
+					database.SaveMetricReturns(errors.New("an error"))
+					go func() {
+						msgChan <- noaa.NewHttpStartStopEnvelope(111111, 100000000, 200000000, 0)
+					}()
+				})
+				It("logs the errors", func() {
+					Consistently(database.SaveMetricCallCount).Should(Equal(0))
+
+					fclock.WaitForWatcherAndIncrement(TestCollectInterval)
+					Eventually(buffer).Should(gbytes.Say("save-metric-to-database"))
+					Eventually(buffer).Should(gbytes.Say("an error"))
+					Eventually(buffer).Should(gbytes.Say("throughput"))
+
+					Eventually(buffer).Should(gbytes.Say("save-metric-to-database"))
+					Eventually(buffer).Should(gbytes.Say("an error"))
+					Eventually(buffer).Should(gbytes.Say("responsetime"))
+				})
+			})
 
 		})
 
-		Context("when there is no containermetric event", func() {
+		Context("when there is no conatinermetrics or httpstartstop event", func() {
```

Comment:
yes, this will be better. 

Container Metrics does not need to compute the metrics,  throughput and responestime do need